### DISCUSSION TITLE
New entities created when updating kiosk URL

### DISCUSSION THREAD

**Discussion by martinnj** [tijsverkoyen/HomeAssistant-FusionSolar#170] [Ideas]
**Describe the bug**
When a the kiosk URL is replaced following the instructions from the README. (https://github.com/tijsverkoyen/HomeAssistant-FusionSolar?tab=readme-ov-file#i-need-to-change-the-kiosk-url) the plugin creates a new entity, breaking every reference to the entities/sensors.

** Debug information **
Answer the questions below before submitting a new issue.

* Are you using the latest version of Home Assistant?
  Yes.
* Are you using the latest version of this integration?
  Yes. 3.0.4
* Have you looked into the closed issues and did not find a similar issue.
  Yes.
* Are you using Kiosk or Northbound / OpenAPI mode?
  Kiosk, our installer has no more open API slots.
* If you are using Kiosk mode. Have you checked that the url is still valid? See [I am using the kiosk mode, but the data is not updating](https://github.com/tijsverkoyen/HomeAssistant-FusionSolar/blob/master/README.md#i-am-using-the-kiosk-mode-but-the-data-is-not-updating)
  Yes, the problem is caused by updating the Kiosk URL.
* If you ar using Northbound / OpenAPI mode. Check if your credentials are still valid. See [How can I verify my Northbound API / OpenAPI credentials?](https://github.com/tijsverkoyen/HomeAssistant-FusionSolar/blob/master/README.md#how-can-i-verify-my-northbound-api--openapi-credentials)
  N/A.
* Have you checked the logs for any info? 
  Yes. There is nothing.

**To Reproduce**
1. Install the plugin using HACS.
2. Configure with Kiosk URL.
3. Rename the entities to remove the "api key" portion and make them humanly parsable.
4. Wait for the URL to expire, or at least generate data.
5. Regenerate the Kiosk URL from the Fusion Solar web portal.
6. Update with new Kiosk URL following the [README](https://github.com/tijsverkoyen/HomeAssistant-FusionSolar?tab=readme-ov-file#i-need-to-change-the-kiosk-url).

**Expected behavior**
When the Kiosk URL is replaced the existing entities should keep producing data instead of new entities being created.

**Screenshots**

**Comment by tijsverkoyen** [comment]
Yeah. I know. This is by design. Won't fix this for now.
I will convert this issue to a discussion. If there are many people who have the same issue, I will reconsider to invest time.

**Reply by martinnj** [reply]
Ahh. I'm guessing it's because the system was really intended for the Northbound/OpenAPI keys which don't change, and you want to be able to store multiple keys to enable multiple sites?

**Reply by MaBeniu** [reply]
This is something very annoying. By using kiosk I was happy until first expiration. Now I see it will not work unless there is a way to update url.
Or of course try getting open API working, which I have some struggles getting response from installer company.
